node/snapshots.rei

```diff
@@ -1,24 +1,43 @@
 open Crypto;
 open Protocol;
 
+type snapshot = {
+  hash: BLAKE2B.t,
+  data: string,
+};
+
+type snapshot_ref;
+
 type t =
   pri {
-    last_snapshot: (BLAKE2B.t, string),
-    last_snapshot_height: int64,
+    current_snapshot: snapshot_ref,
+    next_snapshots: list((int64, snapshot_ref)),
     last_block: Block.t,
     last_block_signatures: Signatures.t,
     additional_blocks: list(Block.t),
   };
 
 let make:
   (
-    ~initial_snapshot: (BLAKE2B.t, string),
+    ~initial_snapshot: snapshot,
     ~initial_block: Block.t,
     ~initial_signatures: Signatures.t
   ) =>
   t;
 
 /** this should be called when a block is received */
 let append_block: (~pool: Block_pool.t, (Block.t, Signatures.t), t) => t;
-let update:
-  (~new_snapshot: (BLAKE2B.t, string), ~applied_block_height: int64, t) => t;
+let add_snapshot_ref: (~block_height: int64, t) => (snapshot_ref, t);
+
+let set_snapshot_ref: (snapshot_ref, snapshot) => unit;
+
+/** Combines add_snapshot_ref and set_snapshot_ref in a single operation.
+    Used when synchronously hashing a protocol state.
+*/
+let add_snapshot: (~block_height: int64, snapshot, t) => t;
```

Comment:
I would prefer if this function was removed, it is effectively only an util as it doesn't wrap considerable logic

Reply:
Done